runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -240,15 +285,55 @@ void registerTable(@Nonnull String namespace,
         }
     }
 
+    /**
+     * For each protobuf filename -> descriptor map, validate if the previous descriptor matches
+     * If there is a new schema or an update, record the new update in the table.
+     * (Note that oldProto means existing schema previously opened)
+     * WARNING: This method MUST be invoked within a transaction.
+     *
+     * @param protoName - name of the protobuf file
+     * @param newProtoFd - descriptor of the protobuf file that is being inserted.
+     * @param namespace - namespace of the table
+     * @param tableName - tablename without the namespace
+     */
+    private void recordNewSchema(ProtobufFileName protoName,
+                                    CorfuRecord<ProtobufFileDescriptor, TableMetadata> newProtoFd,
+                                    String namespace,
+                                    String tableName) {
+        final CorfuRecord<ProtobufFileDescriptor, TableMetadata> oldProto = this.protobufDescriptorTable.get(protoName);
+        if (oldProto == null) {
+            this.protobufDescriptorTable.put(protoName, newProtoFd);
```

Comment:
wouldnt this have already been populated in insertDependingFileDescriptorProtos()?

Reply:
No, in `insertAllDependingFileDescriptorProtos()` we do not touch the actual protobuf Table because we want to do it transactionally.